plugins/local_contents/txl_local_contents/components.py

```diff
@@ -18,21 +18,35 @@ async def get(
         type: str = "unicode",
     ) -> Union[List, Y.YDoc]:
         p = Path(path)
-        assert p.is_dir() == is_dir
-        if p.is_dir():
+        assert (await p.is_dir()) == is_dir
```

Comment:
Switching from `pathlib.Path` to `anyio.Path` makes the application hang on Ctrl-C. One has to press Ctrl-C a second time to stop the application, with this error message:
```
Exception ignored in: <module 'threading' from '/home/david/micromamba/envs/jpterm/lib/python3.11/threading.py'>
Traceback (most recent call last):
  File "/home/david/micromamba/envs/jpterm/lib/python3.11/threading.py", line 1583, in _shutdown
    lock.acquire()
KeyboardInterrupt: 

Aborted!
```
I've been unable to create a minimal reproducible example, but I've seen that as soon as I use `anyio.to_thread.run_sync()`, even to run an empty function, it hangs, so it's not an issue of some code not finished running in a thread. Jpterm doesn't use threads otherwise, so I'm running out of ideas.
@agronholm do you think of something on the top of your head that could explain this behavior?

Reply:
`anyio.to_thread` uses its own threads and not asyncio's thread pool executor, so something there could explain this. I can't point to the actual cause, but this at least should be a clue.

Reply:
I can see that a [thread is stopped when the root task is done](https://github.com/agronholm/anyio/blob/f1f9991fef8a7987eb6d9250de8a4b58d3cbdc2f/src/anyio/_backends/_asyncio.py#L860), but I see that `stop` is not called on my thread. Since jpterm uses Asphalt, maybe this doesn't play well with Asphalt's way of stopping an application with Ctrl-C?